### PR TITLE
Optimized rpc packet size when only the `PackedByteArray` is sent.

### PR DESCRIPTION
**Pull request by AndreaCatania**
This PR is the replacement of #36151 with the @Faless requested changes.

Sample code to test it:
```gdscript
extends Node

func _ready():
	var _self = self
	var args = Array(OS.get_cmdline_args())
	while args.size():
		var a = args.pop_front()
		if a == 'c':
			var peer = NetworkedMultiplayerENet.new()
			peer.create_client("localhost", 8092)
			multiplayer.connect("network_peer_connected", Callable(_self, "_connected"))
			multiplayer.set_network_peer(peer)
		elif a == 's':
			var peer = NetworkedMultiplayerENet.new()
			peer.create_server(8092)
			multiplayer.connect("network_peer_connected", Callable(_self, "_connected"))
			multiplayer.set_network_peer(peer)

	rpc_config("_added", MultiplayerAPI.RPC_MODE_REMOTESYNC)
	rpc_config("_other", MultiplayerAPI.RPC_MODE_REMOTESYNC)


func _pkt(p_id : int, pkt : PackedByteArray):
	if pkt[0] != 0:
		return # Just for testing
	if not multiplayer.is_network_server():
		prints(p_id, pkt.subarray(1, pkt.size() - 1).get_string_from_utf8(), "at", OS.get_ticks_msec())
		multiplayer.send_bytes(pkt, 1)
	else:
		pkt = pkt.subarray(1, pkt.size() - 1)
		print("Recv ping at: %s = %s" % [OS.get_ticks_msec(), OS.get_ticks_msec() - int(pkt.get_string_from_utf8())])
		return
		multiplayer.send_bytes(str(OS.get_ticks_msec()).to_utf8(), p_id)


func _connected(p_id):
	print("Connected to: %d" % p_id)
	if multiplayer.is_network_server():
		while true:
			yield(get_tree().create_timer(1),"timeout")
			if not p_id in multiplayer.get_network_connected_peers():
				break
			print("Sending ping at ", OS.get_ticks_msec())
			var pkt = PackedByteArray([0])
			pkt.append_array(str(OS.get_ticks_msec()).to_utf8())
			multiplayer.send_bytes(pkt, p_id)
			rpc_id(p_id, "_ping", OS.get_ticks_msec())
			rpc("_added", "Added Called")
			var bytes = PackedByteArray()
			bytes.resize(4)
			bytes[0] = 1
			bytes[1] = 2
			bytes[2] = 3
			bytes[3] = 4
			rpc_unreliable_id(p_id, "_other", bytes)

	if multiplayer.is_network_server():
		var ticks = OS.get_ticks_msec()
		print("Ping: %d" % ticks)
		rpc_id(p_id, "_ping", ticks)
		for i in range(0, 10):
			yield(get_tree().create_timer(1),"timeout")
			ticks = OS.get_ticks_msec()
			print("Ping: %d" % ticks)
			rpc_id(p_id, "_ping", ticks)


remote func _ping(time):
	print("Ping: %d" % time)
	rpc_id(1, "_pong", time)

remote func _pong(time):
	print("Pong: %d - %d = %d" % [OS.get_ticks_msec(), time, OS.get_ticks_msec() - time])


func _added(msg):
	print("Added called with: ", msg)


func _other(bytes: PackedByteArray):
	if bytes.size() != 4:
		print("Other call, but the size is wrong")
		return
	print("Other called with bytes: ", bytes[0], bytes[1], bytes[2], bytes[3])

```

This work has been kindly sponsored by IMVU.

### CODE REVIEW

File: core/io/multiplayer_api.cpp [hunk 1]
```diff
@@ -854,7 +905,7 @@ void MultiplayerAPI::_send_rpc(Node *p_from, int p_to, bool p_unreliable, bool p
 	ERR_FAIL_COND(name_id_compression > 1);
 
 	// We can now set the meta
-	packet_cache.write[0] = command_type + (node_id_compression << 3) + (name_id_compression << 5);
+	packet_cache.write[0] = command_type + (node_id_compression << 3) + (name_id_compression << 5) + ((byte_only_or_no_args ? 1 : 0) << 6);
```
Comment (Faless) [inline core/io/multiplayer_api.cpp]: We should probably make an enum with these metadata.

**Review by Faless — APPROVED** [review]
This looks much better. See my only comment, not really a blocker though. Great job! :trophy:

File: core/io/multiplayer_api.cpp (hunk 1 above)
Comment (AndreaCatania) [reply core/io/multiplayer_api.cpp]: Do you mean `<< 3`, `<< 5` ?
Yes, I agree.

Comment (AndreaCatania) [reply core/io/multiplayer_api.cpp]: I've replaced it with a macro.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**